src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/Internal/InMemoryFileInfo.cs

```diff
@@ -8,9 +8,9 @@ namespace Microsoft.Extensions.FileSystemGlobbing.Internal
 {
     internal sealed class InMemoryFileInfo : FileInfoBase
     {
-        private InMemoryDirectoryInfo _parent;
+        private InMemoryDirectoryInfo? _parent;
 
-        public InMemoryFileInfo(string file, InMemoryDirectoryInfo parent)
+        public InMemoryFileInfo(string file, InMemoryDirectoryInfo? parent)
```

Comment:
It looks like there are 2 calls to this constructor, both pass in `this` to `parent`. Why does this need to be nullable?

Reply:
Hm, I think it has something to do with that `ParentDirectory` is nullable. This one was possibly marked by inertia. But here it seems that parent directory always exists. Since the class is internal, it shouldn't be a problem to mark it as nullable if something changes.

Marked it as non-nullable.